server/src/main/java/com/objectcomputing/checkins/services/checkin_notes/CheckinNoteServicesImpl.java

```diff
@@ -1,39 +1,62 @@
 package com.objectcomputing.checkins.services.checkin_notes;
 
+import com.objectcomputing.checkins.services.checkins.CheckIn;
+import com.objectcomputing.checkins.services.checkins.CheckInBadArgException;
 import com.objectcomputing.checkins.services.checkins.CheckInRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
 import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.memberprofile.currentuser.CurrentUserServices;
+import com.objectcomputing.checkins.services.role.RoleType;
+import io.micronaut.security.utils.SecurityService;
+import jnr.a64asm.Mem;
 
 import javax.inject.Inject;
+import javax.inject.Singleton;
 import javax.validation.constraints.NotNull;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
 import java.util.UUID;
 
+@Singleton
 public class CheckinNoteServicesImpl implements CheckinNoteServices {
 
-    @Inject
     private CheckInRepository checkinRepo;
-
-    @Inject
     private CheckinNoteRepository checkinNoteRepository;
-
-    @Inject
     private MemberProfileRepository memberRepo;
+    private SecurityService securityService;
+    private CurrentUserServices currentUserServices;
+
+    public CheckinNoteServicesImpl(CheckInRepository checkinRepo, CheckinNoteRepository checkinNoteRepository,
+                                   MemberProfileRepository memberRepo, SecurityService securityService,
+                                   CurrentUserServices currentUserServices) {
+        this.checkinRepo = checkinRepo;
+        this.checkinNoteRepository = checkinNoteRepository;
+        this.memberRepo = memberRepo;
+        this.securityService = securityService;
+        this.currentUserServices = currentUserServices;
+    }
+
 
     @Override
     public CheckinNote save(CheckinNote checkinNote) {
         CheckinNote checkinNoteRet = null;
+        String workEmail = securityService!=null ? securityService.getAuthentication().get().getAttributes().get("email").toString() : null;
+        MemberProfile currentUser = workEmail!=null? currentUserServices.findOrSaveUser(null, workEmail) : null;
```

Comment:
I thought that was intentional to create a user and not use findByWorkEmail. That what I understood when I was working with @Bagurp on validations. @Bagurp ?

Reply:
That's probably right @meyerinj. I can't think of a scenario where we'd need to create a new user at this point. @bhatiam12, I remember creating a getCurrentUser() in CurrentUserServiceImpl (but removed it for some reason).

What are your thoughts on injecting @SecurityService only on currentUserServiceImpl? i.e, getCurrentUser() will get email address from security service, validate user with memberProfileRepo.findByWorkEmail, getRoles and add all data to CurrentUserDTO and returns it. This way, we've to inject security service in only one service.

I feel like I've been down this road and chose to inject it in every service.

Reply:
I think it's worth us all having a sit down at some point to talk how best to use the SecurityService.

I'm leaning towards thinking we should only use it on the Controller and it will parse out the relevant information and either reject the user there or pass relevant information into the service. I feel like the nature of the server and how authentication is done should be abstracted away from the services. That said I'm not exactly prepared to commit that to an RFC.